test/bundler-integration/module/MultichainValidator.Module.specs.ts

```diff
@@ -0,0 +1,303 @@
+import { expect } from "chai";
+import { ethers, deployments, waffle } from "hardhat";
+import { makeEcdsaModuleUserOp, fillAndSign, makeMultichainEcdsaModuleUserOp } from "../../utils/userOp";
+import { getERC20SessionKeyParams } from "../../utils/sessionKey";
+import { encodeTransfer } from "../../utils/testUtils";
+import { defaultAbiCoder, hexZeroPad, hexConcat } from "ethers/lib/utils";
+import { 
+  getEntryPoint, 
+  getSmartAccountImplementation, 
+  getSmartAccountFactory, 
+  getMockToken, 
+  getEcdsaOwnershipRegistryModule,
+  getVerifyingPaymaster,
+} from "../../utils/setupHelper";
+import {keccak256} from "ethereumjs-util";
+import { MerkleTree } from "merkletreejs";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { BundlerTestEnvironment } from "../environment/bundlerEnvironment";
+
+describe("MultichainValidator Module", async () => {
+
+  const maxAmount = ethers.utils.parseEther("100");
+
+  let [deployer, smartAccountOwner, alice, charlie, verifiedSigner, sessionKey] =
+    [] as SignerWithAddress[];
+
+  let environment: BundlerTestEnvironment;
+
+  before(async function () {
+    const chainId = (await ethers.provider.getNetwork()).chainId;
+    if (chainId !== BundlerTestEnvironment.BUNDLER_ENVIRONMENT_CHAIN_ID) {
+      this.skip();
+    }
+
+    environment = await BundlerTestEnvironment.getDefaultInstance();
+  });
+
+  beforeEach(async function () {
+    [deployer, smartAccountOwner, charlie, verifiedSigner, sessionKey] =
+      await ethers.getSigners();
+  });
+
+  afterEach(async function () {
+    const chainId = (await ethers.provider.getNetwork()).chainId;
+    if (chainId !== BundlerTestEnvironment.BUNDLER_ENVIRONMENT_CHAIN_ID) {
+      this.skip();
+    }
+
+    await Promise.all([
+      environment.revert(environment.defaultSnapshot!),
+      environment.resetBundler(),
+    ]);
+  });
+
+  const setupTests = deployments.createFixture(async ({ deployments, getNamedAccounts }) => {
+
+    // deploy a smart account with a multichain module and enable the session key manager
+    // and a session key all in one userOp
+    await deployments.fixture();
+
+    const entryPoint = await getEntryPoint();
+    const smartAccountFactory = await getSmartAccountFactory();
+    const mockToken = await getMockToken();
+    const ecdsaModule = await getEcdsaOwnershipRegistryModule();
+    
+    const MultichainECDSAValidator = await ethers.getContractFactory("MultichainECDSAValidator");
+    const multichainECDSAValidator = await MultichainECDSAValidator.deploy();
+    const sessionKeyManager = await (await ethers.getContractFactory("SessionKeyManager")).deploy();
```

Comment:
are these being used in this test case anywhere?

Reply:
yes. they are passed to cases from `setupTests`